exec_process.go

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+// startExecCmd starts the exec command and registers the process to the reaper.
+// This function is a helper for exec.Cmd.Start() since this needs to be in sync
+// with exec.Cmd.Wait().
+func (s *sandbox) startExecCmd(c *exec.Cmd) error {
+	s.procChLock.Lock()
+	defer s.procChLock.Unlock()
+
+	if err := c.Start(); err != nil {
+		return err
+	}
+
+	s.processesCh[c.Process.Pid] = make(chan int, 1)
+
+	return nil
+}
+
+// waitProcess blocks until the expected process has been reaped. After the
+// reaping from the subreaper, the exit code is sent through the provided
+// channel.
+// This function is a helper for exec.Cmd.Wait() and os.Process.Wait() since
+// both cannot be used directly, because of the subreaper.
+func (s *sandbox) waitProcess(obj interface{}) (int, error) {
```

Comment:
please define an interface with the Wait() as its function.

Reply:
done